### PR TITLE
feat: Make MNOTE and PUNCH outputs available from VSCode

### PR DESCRIPTION
**Pull request by slavek-kucera**

### CODE REVIEW

File: parser_library/src/workspace_manager.cpp [hunk 1]
```diff
@@ -319,14 +319,19 @@ class workspace_manager_impl final : public workspace_manager,
 
         std::chrono::duration<double> duration = std::chrono::steady_clock::now() - start;
 
-        const auto& [url, metadata, perf_metrics, errors, warnings] = task.value();
+        const auto& [url, metadata, perf_metrics, errors, warnings, outputs_changed] = task.value();
```
Comment (jirimosinger) [inline parser_library/src/workspace_manager.cpp:322]: It's almost asking for a struct as a return value now 😄

File: language_server/test/lsp/feature_language_features_test.cpp [hunk 2]
```diff
@@ -495,3 +495,32 @@ TEST(language_features, folding)
...
+        { "level", -1 },
+        { "text", "A" },
+    });
+    response.push_back({
+        { "level", 2 },
+        { "text", "B" },
+    });
+
+    EXPECT_CALL(response_mock, respond(request_id(0), std::string(""), response));
+    notifs["textDocument/$/retrieve_outputs"].as_request_handler()(request_id(0), params1);
```
Comment (jirimosinger) [inline language_server/test/lsp/feature_language_features_test.cpp:523]: Consider adding test for empty response (e.g. when neither `PUNCH` nor `MNOTE` are used)

File: clients/vscode-hlasmplugin/src/hlasmOutputContentProvider.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+    return result as Options;
+}
+
+function createOutputUri(uri: vscode.Uri, options: Options) {
+    const query = uriFriendlyBase16Encode(uri.toString());
+    const path = `/${translateOptions(options)}/${uri.path.substring(uri.path.lastIndexOf('/') + 1)}`;
+
+    return vscode.Uri.from({ scheme, path, query });
+
```
Comment (jirimosinger) [inline clients/vscode-hlasmplugin/src/hlasmOutputContentProvider.ts:52]:
```suggestion
```

File: clients/vscode-hlasmplugin/src/hlasmOutputContentProvider.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+import { pickUser } from './uiUtils';
+
+type OutputResult = {
+    level: number;
+    text: string;
+}[];
+
+type Options = { mnote: true, punch: true } | { mnote: false, punch: true } | { mnote: true, punch: false };
+
+const scheme = 'hlasm-output';
```
Comment (jirimosinger) [inline clients/vscode-hlasmplugin/src/hlasmOutputContentProvider.ts:28]: I don't remember - do we already store individual schemes somewhere in "scheme storage utility"? And if we don't, does it make sense to start doing now?

File: clients/vscode-hlasmplugin/src/hlasmOutputContentProvider.ts
Comment (jirimosinger) [inline clients/vscode-hlasmplugin/src/hlasmOutputContentProvider.ts:1]: Consider aligning the prefixes so the texts are not "all over the place":

image

Comment (jirimosinger) [inline clients/vscode-hlasmplugin/src/hlasmOutputContentProvider.ts:1]: Let's say I have the following code:

```
         MNOTE 0,'mnote test'
```

This correctly generates the output.

However, when I delete the line, the output stays the same. Is it possible for the generated output to be empty in such a case?

**Review by jirimosinger — COMMENTED** [review]
Definitely not relevant to this specific PR but we should probably unify the upper/lower variants case of the command texts (maybe though some utility function to be safe?):

image

File: parser_library/src/workspace_manager.cpp (hunk 1 above)
Comment (slavek-kucera) [reply parser_library/src/workspace_manager.cpp:322]: There is one, it's just been practical to immediately use structured binding.

File: clients/vscode-hlasmplugin/src/hlasmOutputContentProvider.ts
Comment (slavek-kucera) [reply clients/vscode-hlasmplugin/src/hlasmOutputContentProvider.ts:1]: That looks like a bug... I'll look into it.

Comment (jirimosinger) [inline clients/vscode-hlasmplugin/src/hlasmOutputContentProvider.ts:1]: One thing I've noticed just now is that there is an empty "editor" after reloading VS Code. Is there something we can do about it? Possible closing all "Generated output" windows before VS code exits?

image

**Review by jirimosinger — APPROVED** [review]

File: clients/vscode-hlasmplugin/src/hlasmOutputContentProvider.ts
Comment (jirimosinger) [reply clients/vscode-hlasmplugin/src/hlasmOutputContentProvider.ts:1]: As discussed - probably bug in VS Code
